src/NuGet.Core/NuGet.PackageManagement/Utility/MSBuildNuGetProjectSystemUtility.cs

```diff
@@ -9,16 +9,36 @@
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Xml.Linq;
 using NuGet.Common;
 using NuGet.Frameworks;
 using NuGet.PackageManagement;
 using NuGet.Packaging;
 using NuGet.Packaging.Core;
+using NuGet.Shared;
 
 namespace NuGet.ProjectManagement
 {
     public static class MSBuildNuGetProjectSystemUtility
     {
+        public static XDocument GetOrCreateXmlDocument(XName rootName, string path, IMSBuildProjectSystem msBuildNuGetProjectSystem)
```

Comment:
Can this be made internal?  If not, I think I'd prefer that `NuGet.ProjectManagement.XmlUtility` is not deleted, but instead changed to be a wrapper of `NuGet.Shared.XmlUtility`, if that helps not needing this new public API.

Reply:
Good catch Andy. Changed the access modifier of `MSBuildNuGetProjectSystemUtility.GetOrCreateXmlDocument` method to `internal`.